app/scripts/controllers/deploymentConfig.js

```diff
@@ -56,18 +56,43 @@ angular.module('openshiftConsole')
     var orderByDate = $filter('orderObjectsByDate');
     var mostRecent = $filter('mostRecent');
 
-    // copy deploymentConfig and ensure it has env so that we can edit env vars using key-value-editor
-    var copyDeploymentConfigAndEnsureEnv = function(deploymentConfig) {
-      $scope.updatedDeploymentConfig = angular.copy(deploymentConfig);
-      _.each($scope.updatedDeploymentConfig.spec.template.spec.containers, function(container) {
-        container.env = container.env || [];
-        // check valueFrom attribs and set an alt text for display if present
-        _.each(container.env, function(env) {
-          $filter('altTextForValueFrom')(env);
-        });
-      });
+    var previousEnvConflict = false;
+    var updateEnvironment = function(current, previous) {
+      if (previousEnvConflict) {
+        return;
+      }
+
+      if (!$scope.forms.dcEnvVars || $scope.forms.dcEnvVars.$pristine) {
+        $scope.updatedDeploymentConfig = EnvironmentService.copy(current);
+        return;
+      }
+
+      // The env var form has changed and the deployment config has been
+      // updated. See if there were any background changes to the environment
+      // variables. If not, merge the environment edits into the updated
+      // deployment config object.
+      if (EnvironmentService.isEnvironmentEqual(current, previous)) {
+        $scope.updatedDeploymentConfig = EnvironmentService.mergeEdits($scope.updatedDeploymentConfig, current);
+        return;
+      }
+
+      previousEnvConflict = true;
+      $scope.alerts["background_update"] = {
+        type: "warning",
+        message: "This deployment configuration has been updated in the background. Saving your changes may create a conflict or cause loss of data.",
```

Comment:
now that we know the env vars are what caused the conflict, i'm thinking we should update the message to be more specific